[DOC] add datasource documentation for OpenShift (#422)
perses/perses-operator: config/samples/openshift/README.md

> [!IMPORTANT]
> If you face any issues connecting to Loki, ensure that your Perses instance can reach the `openshift-logging` namespace services and that the token has the necessary permissions (`cluster-logging-application-view`, `cluster-logging-infrastructure-view`, or `cluster-logging-audit-view` cluster roles) to access the log data.

> [!NOTE]
> A sample `PersesGlobalDatasource` configuration is available in `loki-global-datasource.yaml`. It connects to the `application` tenant by default.

### Connect Perses to OpenShift Distributed Tracing (Tempo)

OpenShift Distributed Tracing with Tempo provides a backend for collecting and querying distributed traces. You can configure Perses to query traces from Tempo using the Tempo Datasource plugin.

> [!IMPORTANT]
> **Prerequisites**: The Tempo operator must be installed and a `TempoStack` or `TempoMonolithic` instance with multi-tenancy must be deployed.

1. Login to your OpenShift cluster using the `oc` CLI tool.
2. Retrieve the Tempo gateway service URL. Example URL:

   ```
   https://tempo-simplest-gateway.openshift-tracing.svc.cluster.local:8080/api/traces/v1/<tenant>/tempo
   ```

>    Replace `<tenant>` with the name of your tracing tenant. The tenant name depends on your TempoStack configuration (e.g., `platform`, `dev`). Check your TempoStack CR for the configured tenants:

Reviewer: ```suggestion
   Replace `<tenant>` with the name of your tracing tenant. The tenant name depends on your TempoStack configuration. Check your TempoStack CR for the configured tenants:
```

`platform`/`dev` were just example tenants in our sample manifests, not sure if it makes sense to add them here explicitly

   ```bash
   oc get -n <namespace> tempostack <instance> -o jsonpath='{.items[*].spec.tenants}'
   ```

3. Create a Perses project or use an existing one.
4. In your Perses project, create a new Datasource with the following configuration:
   - **Name**: Tempo Datasource
   - **Set as Default**: true
   - **Plugin Options**: Source: Tempo Datasource
   - **HTTP Settings**: Select Proxy and set the URL to the Tempo gateway URL (from step 2), for example:
     `https://tempo-simplest-gateway.openshift-tracing.svc.cluster.local:8080/api/traces/v1/tenant1/tempo`

>    - **Headers**: Add the header `X-Scope-OrgID` with the value matching your tracing tenant (e.g., `dev`)

Reviewer: ```suggestion
```

not required, as the tenant name is already in the URL

   - **Secret**: Set to `<datasource-name>-secret`. The operator automatically creates this secret; the proxy uses it for authentication. For example, if the datasource name is `tempo-datasource`, set the secret to `tempo-datasource-secret`.
   - Save the Datasource.
5. In your Perses project, create a Dashboard and add Panels with trace queries. As the Datasource is the default, all trace queries will use the Tempo datasource by default.

> [!IMPORTANT]
> If you face any issues connecting to Tempo, ensure that your Perses instance can reach the Tempo instance, that the Tempo gateway is accessible, and that the logged in user has the necessary permissions to access tracing data.

> [!NOTE]

> > A sample `PersesGlobalDatasource` configuration is available in `tempo-global-datasource.yaml`. It connects to the `platform` tenant with `X-Scope-OrgID: dev` by default.

Reviewer: ```suggestion
> A sample `PersesGlobalDatasource` configuration is available in `tempo-global-datasource.yaml`. It connects to the `tenant1` tenant by default.
```

   oc get route thanos-querier -n openshift-monitoring -o jsonpath='{.spec.host}'
   ```
3. Create a Perses project or use an existing one.
4. In your Perses project, create a new Datasource with the following configuration:
   - **Name**: Openshift Thanos Querier
   - **Set as Default**: true
   - **Plugin Options**: Source: Prometheus Datasource
   - **HTTP Settings**: Select Proxy and set the URL to `https://<thanos-querier-route-url>` (obtained from step 2)

>    - **Secret**: Set to `<datasource-name>-secret`. The operator automatically creates this secret; the proxy uses it for authentication. For example, if the datasource name is `thanos-querier-datasource`, set the secret to `thanos-querier-datasource-secret`.

Reviewer: Can we automate this step / infer the secret name?

Author: We could. I just wanted to document the current status.

   - Save the Datasource.
5. In your Perses project, create a Dashboard and add Panels with log queries. As the Datasource is the default, all log queries will use the Loki datasource by default.

> [!IMPORTANT]
> If you face any issues connecting to Loki, ensure that your Perses instance can reach the `openshift-logging` namespace services and that the token has the necessary permissions (`cluster-logging-application-view`, `cluster-logging-infrastructure-view`, or `cluster-logging-audit-view` cluster roles) to access the log data.

> [!NOTE]
> A sample `PersesGlobalDatasource` configuration is available in `loki-global-datasource.yaml`. It connects to the `application` tenant by default.

### Connect Perses to OpenShift Distributed Tracing (Tempo)

OpenShift Distributed Tracing with Tempo provides a backend for collecting and querying distributed traces. You can configure Perses to query traces from Tempo using the Tempo Datasource plugin.

> [!IMPORTANT]

> > **Prerequisites**: OpenShift Distributed Tracing (Tempo) must be deployed in the `openshift-tracing` namespace. Verify by running:

Reviewer: ```suggestion
> **Prerequisites**: The Tempo operator must be installed and a `TempoStack` or `TempoMonolithic` instance with multi-tenancy must be deployed.
```

>
> ```bash
> oc get pods -n openshift-tracing

> > ```

Reviewer: ```suggestion
```

1. Login to your OpenShift cluster using the `oc` CLI tool.

> 2. Retrieve the Tempo gateway service URL. The default in-cluster URL is:

Reviewer: ```suggestion
2. Retrieve the Tempo gateway service URL. Example URL:
```

   ```
   https://tempo-simplest-gateway.openshift-tracing.svc.cluster.local:8080/api/traces/v1/<tenant>/tempo
   ```

   Replace `<tenant>` with the name of your tracing tenant. The tenant name depends on your TempoStack configuration (e.g., `platform`, `dev`). Check your TempoStack CR for the configured tenants:

   ```bash

>    oc get tempostack -n openshift-tracing -o jsonpath='{.items[*].spec.tenants}'

Reviewer: ```suggestion
   oc get -n <namespace> tempostack <instance> -o jsonpath='{.items[*].spec.tenants}'
```

   ```

3. Create a Perses project or use an existing one.
4. In your Perses project, create a new Datasource with the following configuration:
   - **Name**: Tempo Datasource
   - **Set as Default**: true
   - **Plugin Options**: Source: Tempo Datasource
   - **HTTP Settings**: Select Proxy and set the URL to the Tempo gateway URL (from step 2), for example:

>      `https://tempo-simplest-gateway.openshift-tracing.svc.cluster.local:8080/api/traces/v1/platform/tempo`

Reviewer: ```suggestion
     `https://tempo-simplest-gateway.openshift-tracing.svc.cluster.local:8080/api/traces/v1/tenant1/tempo`
```

   - **Headers**: Add the header `X-Scope-OrgID` with the value matching your tracing tenant (e.g., `dev`)
   - **Secret**: Set to `<datasource-name>-secret`. The operator automatically creates this secret; the proxy uses it for authentication. For example, if the datasource name is `tempo-datasource`, set the secret to `tempo-datasource-secret`.
   - Save the Datasource.
5. In your Perses project, create a Dashboard and add Panels with trace queries. As the Datasource is the default, all trace queries will use the Tempo datasource by default.

> [!IMPORTANT]

> > If you face any issues connecting to Tempo, ensure that your Perses instance can reach the `openshift-tracing` namespace services, that the Tempo gateway is accessible, and that the token has the necessary permissions to access tracing data.

Reviewer: ```suggestion
> If you face any issues connecting to Tempo, ensure that your Perses instance can reach the Tempo instance, that the Tempo gateway is accessible, and that the logged in user has the necessary permissions to access tracing data.
```
